adam-core/src/main/scala/org/bdgenomics/adam/rdd/GenomicRDD.scala

```diff
@@ -757,18 +757,21 @@ trait GenomicRDD[T, U <: GenomicRDD[T, U]] extends Logging {
    * @see broadcastRegionJoinAgainst
    */
   def broadcastRegionJoin[X, Y <: GenomicRDD[X, Y], Z <: GenomicRDD[(T, X), Z]](
-    genomicRdd: GenomicRDD[X, Y])(
+    genomicRdd: GenomicRDD[X, Y],
+    threshold: Long = 0L)(
```

Comment:
`threshold` param needs scaladoc here and elsewhere.

`flankSize`, `padding`, and `threshold` all have similar meanings, is there any reason to choose one over the other?

Finally, adding a parameter with a Scala default value may make life more difficult for the Java API and downstream in Python and R.

Reply:
Thanks for the scaladoc reminder. I need to push the tests too.

I'm not tied to `threshold`, but for a join, `padding` and `flankSize` don't seem to capture the meaning of the input: to find nearby regions within some threshold.

I can rewrite it to separate the optional parameter into two different methods if that's better.

Reply:
+1 @heuermh RE consistency. My preference is `flankSize`, as that is the amount of flanking sequence to consider when running an operation. Not a huge deal one way or the other, as long as we're consistent.

>> Finally, adding a parameter with a Scala default value may make life more difficult for the Java API and downstream in Python and R.
>
> I can rewrite it to separate the optional parameter into two different methods if that's better.

+1 @heuermh and +1 at separating into two methods

Reply:
Now sometimes we have four
```scala
def join(GenomicRDD)
def join(GenomicRDD, Option[Int])
def join(GenomicRDD, Long)
def join(GenomicRDD, Option[Int], Long)
```

I think it is ok to have only the simplest and longest forms, especially if one is an option
```scala
def join(GenomicRDD)
def join(GenomicRDD, Option[Int], Long)
```

Reply:
I have mixed feelings on that. I don't think 4 is too many, but I do see the issue. I don't know if we should restrict users in that way.

Related: How do you pass an `Option` from R/Python? 

We may just keep the method signatures with the `optPartitions` parameter `private[adam]` (because hopefully in the future we can deterministically set the partition number) and go with `genomicRDD: GenomicRDD, flankSize: Long` for the public API. Do users need the power to set the partition count?

Reply:
>  How do you pass an Option from R/Python?

From Java I import `scala.Option`/`scala.Some` directly.

I found a library that did Java `Optional` &rarr; Scala `Option` conversions but didn't feel it was worth adding another dependency.

Reply:
Here is what I am proposing:

```scala
public def join(GenomicRDD)
public def join(GenomicRDD, Long)
private[rdd] def join(GenomicRDD, Option[Int], Long)
```

I just don't see users caring about the number of partitions for their join, and we can probably do better deterministically setting it.

Reply:
`public` specified for clarity.

Reply:
That SGTM.